src/txn/txn.c

```diff
@@ -864,7 +901,7 @@ __wt_txn_commit(WT_SESSION_IMPL *session, const char *cfg[])
 		WT_ASSERT(session,
 		    txn->commit_timestamp <= txn->durable_timestamp);
 
-	WT_ERR(__txn_commit_timestamps_assert(session));
+	WT_ERR(__txn_commit_timestamps_assert(session, &cursors));
```

Comment:
I am unable to follow the change. We do not use "cursors" anywhere after this point and free at the end of the function. Can we in that case instead just have cursors as a local variable in {{__txn_commit_timestamps_assert}}. I am clearly missing something here.

Reply:
The trouble comes from this part here: https://github.com/wiredtiger/wiredtiger/blob/develop/src/txn/txn.c#L735
So the cursors need to stay open for any subsequent traversal of the update lists. I'm going to discuss with @agorrod later as maybe there's a way to avoid this.